File: erlang/p_11.erl
-module(p_11).
-import(p_10, [encode/1]).
-export([encode_modify/1]).
-import(lists, [map/2]).


encode_modify(L) ->
    map(fun ([N, E]) -> if 1 =:= N -> E;
			   true -> [N, E]
			end
	end, encode(L)).

File: prolog/p-11.pl
:- ensure_loaded('p-10').

transform([1, E], E) :- !.
transform([N, E], [N, E]) :- !.

encode_modify(Xs, Ys) :-
    encode(Xs, Zs),
    maplist(transform, Zs, Ys).

